godot-core/src/engine/gfile.rs

```diff
@@ -0,0 +1,823 @@
+    /// For more information about underlying method see
+    /// [Godot documentation](https://docs.godotengine.org/en/stable/classes/class_fileaccess.html#class-fileaccess-method-get-64).
+    #[doc(alias("get_64"))]
+    pub fn read_u64(&mut self) -> IoResult<u64> {
+        let val = self.fa.get_64();
+        self.check_error()?;
+        Ok(val)
+    }
+
+    /// Reads the Pascal String from file as [`GString`].
+    ///
+    /// Pascal String is useful for writing and retrieving verying-length string
+    /// data from binary files. It is saved with length-prefix, instead of null
+    /// terminator as in C strings. Text is interpreted as being UTF-8 encoded.
+    ///
+    /// See also:
+    /// - [Wikipedia article](https://en.wikipedia.org/wiki/String_(computer_science)#Length-prefixed)
+    /// - [Godot documentation](https://docs.godotengine.org/en/stable/classes/class_fileaccess.html#class-fileaccess-method-get-pascal-string)
+    #[doc(alias("get_pascal_string"))]
+    pub fn read_pascal_string(&mut self) -> IoResult<GString> {
+        let val = self.fa.get_pascal_string();
+        self.check_error()?;
+        Ok(val)
+    }
+
+    /// Reads the whole file as [`GString`].
+    ///
+    /// If `skip_cr` is set to `true`, carriage returns will be ingored when
+    /// parsing and only line feed characters will represents a new line. Text
+    /// is interpreted as being UTF-8 encoded.
+    ///
+    /// To retrieve the file as [`String`] instead, use [GFile::read_to_string()](https://doc.rust-lang.org/std/io/trait.Read.html#method.read_to_string)
+    /// [`Read`] trait method.
+    ///
+    /// For more information about underlying method see
+    /// [Godot documentation](https://docs.godotengine.org/en/stable/classes/class_fileaccess.html#class-fileaccess-method-get-as-text).
+    #[doc(alias("get_as_text"))]
+    pub fn read_as_gstring_entire(&mut self, skip_cr: bool) -> IoResult<GString> {
+        let val = self.fa.get_as_text_ex().skip_cr(skip_cr).done();
+        self.check_error()?;
+        Ok(val)
+    }
+
+    /// Reads the next line of the file in delimiter-separated file.
+    ///
+    /// For reading traditional `CSV` format, provide comma (`','`) as `delim`.
+    ///
+    /// For more information about underlying method see
+    /// [Godot documentation](https://docs.godotengine.org/en/stable/classes/class_fileaccess.html#class-fileaccess-method-get-csv-line).
+    #[doc(alias("get_csv_line"))]
+    pub fn read_csv_line(&mut self, delim: impl Into<GString>) -> IoResult<PackedStringArray> {
+        let val = self.fa.get_csv_line_ex().delim(delim.into()).done();
+        self.check_error()?;
+        Ok(val)
+    }
+
+    /// Reads the next 32 bits from file as [`f32`].
+    ///
+    /// For more information about underlying method see
+    /// [Godot documentation](https://docs.godotengine.org/en/stable/classes/class_fileaccess.html#class-fileaccess-method-get-real).
+    #[doc(alias("get_real", "get_float"))]
```

Comment:
Is `get_real` not depending on whether Godot is compiled in single- or double-precision format?